src/app/login/login.component.ts

```diff
@@ -50,7 +50,14 @@ export class LoginComponent implements OnInit {
                     this.ngProgress.done();
                     if (error.status === 0) {
                         console.log('service down', error);
-                    } else {
+                    } else if (error.status === 401) {
+                        if (error.statusText.toUpperCase().indexOf('PASSWORD') >= 0 && error.statusText.toUpperCase().indexOf('EXPIRED') >= 0) {
+                            console.log("text", error.statusText);
+                            this.router.navigate(['/reset-password'], { queryParams: { username: this.model.username } });
+                            this.alertService.error(error.statusText);
```

Comment:
keep this out of `if (error.statusText.toUpperCase()` block to know what is happening